tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -58,44 +117,26 @@ ROOT::Experimental::RNTupleChainProcessor::RNTupleChainProcessor(const std::vect
          fEntry->BindValue(token, valuePtr);
       }
 
-      fFieldContexts.emplace_back(field.Clone(field.GetFieldName()), token);
+      const auto &[fieldContext, _] = fFieldContexts.try_emplace(
+         field.GetFieldName(), field.Clone(field.GetFieldName()), token, fPageSource->GetNTupleName());
+      ConnectField(fieldContext->second, *fPageSource, *fEntry);
    }
-
-   ConnectFields();
 }
 
 ROOT::Experimental::NTupleSize_t
 ROOT::Experimental::RNTupleChainProcessor::ConnectNTuple(const RNTupleSourceSpec &ntuple)
```

Comment:
I would add some comment in the function about the logic used. There are two for-loops on the `fFieldContexts` contents very close to each other and it's not immediately clear why